lib/core/test/element_map_to_bins_test.cpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+#include "scipp/core/element/map_to_bins.h"
+
+#include <algorithm>
+#include <gtest/gtest.h>
+#include <iterator>
+#include <random>
+#include <vector>
+
+using namespace scipp;
+using namespace scipp::core::element;
+
+auto random_shuffled(const scipp::index nevent, const scipp::index nbin) {
+  std::random_device rnd_device;
+  std::mt19937 mersenne_engine{rnd_device()};
+  std::uniform_int_distribution<scipp::index> dist{0, nbin - 1};
+  auto gen = [&dist, &mersenne_engine]() { return dist(mersenne_engine); };
+  std::vector<scipp::index> vec(nevent);
+  std::generate(begin(vec), end(vec), gen);
+  std::shuffle(begin(vec), end(vec), mersenne_engine);
```

Comment:
What does shuffling do? Is the vector not already random?

Reply:
Erm, you are right, I think I couldn't decide between two ways of generating the test data and ended up convoluting both. :D